### PR TITLE
Curtain and Flap (De)construction

### PR DESCRIPTION
**Pull request by Doxxmedearly**
Plastic flaps and curtains can now be cut down.
Curtains begin anchored in place and can be anchored/unanchored with a screwdriver.
Curtains (except "plastic curtains")  can be constructed from 2 cloth sheets, giving cloth an actual purpose.
Cutting down a curtain with wirecutters returns both sheets (or, if a plastic curtain, 1 plastic sheet). Using a knife or something returns one (or none if plastic curtain).

### CODE REVIEW

**Comment by Doxxmedearly** [comment]
!review

**Comment by alsoandanswer** [comment]
personally, i think var/inuse would be less verbal diarrhea than var/manipulating

File: code/game/objects/structures/curtains.dm [hunk 1]
```diff
@@ -30,6 +32,49 @@
...
+		visible_message(span("notice", "[user] begins cutting down \the [src]."),
+					span("notice", "You begin cutting down \the [src]."))
+		if(!do_after(user, 30/W.toolspeed))
+			manipulating = FALSE
+			return
+		playsound(src.loc, 'sound/items/Wirecutter.ogg', 50, 1)
+		visible_message(span("notice", "[user] cuts down \the [src]."),
+					span("notice", "You cut down \the [src]."))
+		if(istype(src, /obj/structure/curtain/open/medical))
+			new/obj/item/stack/material/plastic(src.loc)
```
Comment (NonQueueingMatt) [inline code/game/objects/structures/curtains.dm]:
```suggestion
			new /obj/item/stack/material/plastic(src.loc)
```

File: code/game/objects/structures/curtains.dm [hunk 2]
```diff
@@ -30,6 +32,49 @@
...
+		else
+			new/obj/item/stack/material/cloth(src.loc, 2)
```
Comment (NonQueueingMatt) [inline code/game/objects/structures/curtains.dm]:
```suggestion
			new /obj/item/stack/material/cloth(src.loc, 2)
```

File: code/game/objects/structures/curtains.dm [hunk 3]
```diff
@@ -30,6 +32,49 @@
...
+		visible_message(span("notice", "[user] begins cutting down \the [src]."),
+					span("notice", "You begin cutting down \the [src]."))
+		if(!do_after(user, 30))
+			manipulating = FALSE
+			return
+		playsound(src.loc, pick('sound/items/rip1.ogg','sound/items/rip2.ogg'), 50, 1)
+		visible_message(span("notice", "[user] cuts down \the [src]."),
+					span("notice", "You cut down \the [src]."))
+		if(!istype(src, /obj/structure/curtain/open/medical))
+			new/obj/item/stack/material/cloth(src.loc, 1)
```
Comment (NonQueueingMatt) [inline code/game/objects/structures/curtains.dm]:
```suggestion
			new /obj/item/stack/material/cloth(src.loc, 1)
```

File: code/modules/materials/material_recipes.dm [hunk 4]
```diff
@@ -155,6 +155,19 @@
 		new/datum/stack_recipe("yellow folder", /obj/item/folder/yellow), \
 		))
 
+/material/cloth/generate_recipes()
+	..()
+	recipes += new/datum/stack_recipe_list("curtains list",list( \
+		new/datum/stack_recipe("white curtain", /obj/structure/curtain, 2, time = 10), \
```
Comment (NonQueueingMatt) [inline code/modules/materials/material_recipes.dm]:
```suggestion
		new /datum/stack_recipe("white curtain", /obj/structure/curtain, 2, time = 10), \
```

File: code/modules/materials/material_recipes.dm [hunk 5]
```diff
@@ -155,6 +155,19 @@
 		new/datum/stack_recipe("yellow folder", /obj/item/folder/yellow), \
 		))
 
+/material/cloth/generate_recipes()
+	..()
+	recipes += new/datum/stack_recipe_list("curtains list",list( \
+		new/datum/stack_recipe("white curtain", /obj/structure/curtain, 2, time = 10), \
+		new/datum/stack_recipe("bed curtain", /obj/structure/curtain/open/bed, 2, time = 10), \
```
Comment (NonQueueingMatt) [inline code/modules/materials/material_recipes.dm]:
```suggestion
		new /datum/stack_recipe("bed curtain", /obj/structure/curtain/open/bed, 2, time = 10), \
```

File: code/modules/materials/material_recipes.dm [hunk 6]
```diff
@@ -155,6 +155,19 @@
 		new/datum/stack_recipe("yellow folder", /obj/item/folder/yellow), \
 		))
 
+/material/cloth/generate_recipes()
+	..()
+	recipes += new/datum/stack_recipe_list("curtains list",list( \
+		new/datum/stack_recipe("white curtain", /obj/structure/curtain, 2, time = 10), \
+		new/datum/stack_recipe("bed curtain", /obj/structure/curtain/open/bed, 2, time = 10), \
+		new/datum/stack_recipe("black curtain", /obj/structure/curtain/black, 2, time = 10), \
```
Comment (NonQueueingMatt) [inline code/modules/materials/material_recipes.dm]:
```suggestion
		new /datum/stack_recipe("black curtain", /obj/structure/curtain/black, 2, time = 10), \
```

File: code/modules/materials/material_recipes.dm [hunk 7]
```diff
@@ -155,6 +155,19 @@
 		new/datum/stack_recipe("yellow folder", /obj/item/folder/yellow), \
 		))
 
+/material/cloth/generate_recipes()
+	..()
+	recipes += new/datum/stack_recipe_list("curtains list",list( \
+		new/datum/stack_recipe("white curtain", /obj/structure/curtain, 2, time = 10), \
+		new/datum/stack_recipe("bed curtain", /obj/structure/curtain/open/bed, 2, time = 10), \
+		new/datum/stack_recipe("black curtain", /obj/structure/curtain/black, 2, time = 10), \
+		new/datum/stack_recipe("shower curtain", /obj/structure/curtain/open/shower, 2, time = 10), \
```
Comment (NonQueueingMatt) [inline code/modules/materials/material_recipes.dm]:
```suggestion
		new /datum/stack_recipe("shower curtain", /obj/structure/curtain/open/shower, 2, time = 10), \
```

File: code/modules/materials/material_recipes.dm [hunk 8]
```diff
@@ -155,6 +155,19 @@
...
 		))
 
+/material/cloth/generate_recipes()
+	..()
+	recipes += new/datum/stack_recipe_list("curtains list",list( \
+		new/datum/stack_recipe("white curtain", /obj/structure/curtain, 2, time = 10), \
+		new/datum/stack_recipe("bed curtain", /obj/structure/curtain/open/bed, 2, time = 10), \
+		new/datum/stack_recipe("black curtain", /obj/structure/curtain/black, 2, time = 10), \
+		new/datum/stack_recipe("shower curtain", /obj/structure/curtain/open/shower, 2, time = 10), \
+		new/datum/stack_recipe("orange shower curtain", /obj/structure/curtain/open/shower/engineering, 2, time = 10), \
```
Comment (NonQueueingMatt) [inline code/modules/materials/material_recipes.dm]:
```suggestion
		new /datum/stack_recipe("orange shower curtain", /obj/structure/curtain/open/shower/engineering, 2, time = 10), \
```

File: code/modules/materials/material_recipes.dm [hunk 9]
```diff
@@ -155,6 +155,19 @@
...
+		new/datum/stack_recipe("red shower curtain", /obj/structure/curtain/open/shower/security, 2, time = 10), \
```
Comment (NonQueueingMatt) [inline code/modules/materials/material_recipes.dm]:
```suggestion
		new /datum/stack_recipe("red shower curtain", /obj/structure/curtain/open/shower/security, 2, time = 10), \
```

File: code/modules/materials/material_recipes.dm [hunk 10]
```diff
@@ -155,6 +155,19 @@
...
+		new/datum/stack_recipe("privacy curtain", /obj/structure/curtain/open/privacy, 2, time = 10), \
```
Comment (NonQueueingMatt) [inline code/modules/materials/material_recipes.dm]:
```suggestion
		new /datum/stack_recipe("privacy curtain", /obj/structure/curtain/open/privacy, 2, time = 10), \
```

**Comment by fernerr** [comment]
I'd like if medical curtains specifically remained unanchored, it's helpful to be able to move them easily during certain circumstances.

**Comment by Doxxmedearly** [comment]
I suppose medical doesn't have easy access to screwdrivers. That should be easy enough to do when I'm home.

File: code/game/objects/structures/curtains.dm [hunk 11]
```diff
@@ -30,6 +32,49 @@
...
+					span("notice", "You begin cutting down \the [src]."))
+		if(!do_after(user, 30))
+			manipulating = FALSE
+			return
+		playsound(src.loc, pick('sound/items/rip1.ogg','sound/items/rip2.ogg'), 50, 1)
+		visible_message(span("notice", "[user] cuts down \the [src]."),
+					span("notice", "You cut down \the [src]."))
+		if(!istype(src, /obj/structure/curtain/open/medical))
+			new /obj/item/stack/material/cloth(src.loc, 1)
+		qdel(src)
```
Comment (NonQueueingMatt) [inline code/game/objects/structures/curtains.dm]: This entire block is duplicate code and can definitely be condensed into the former one

File: code/game/objects/structures/curtains.dm [hunk 12]
```diff
@@ -46,11 +91,13 @@
 /obj/structure/curtain/medical
 	name = "plastic curtain"
 	color = "#B8F5E3"
+	anchored = FALSE
```
Comment (NonQueueingMatt) [inline code/game/objects/structures/curtains.dm]: this is not changelogged

**Pull request merged**